stl/inc/mutex

```diff
@@ -71,6 +90,10 @@ private:
     _Mtx_t _Mymtx() noexcept { // get pointer to _Mtx_internal_imp_t inside _Mtx_storage
         return reinterpret_cast<_Mtx_t>(&_Mtx_storage);
     }
+
+    int& _Ownership_levels() noexcept { // the count is denoted by an int subobject at the end of _Mtx_storage
+        return *reinterpret_cast<int*>(reinterpret_cast<unsigned char*>(&_Mtx_storage + 1) - sizeof(int));

```

Comment:
```suggestion
        return *(reinterpret_cast<int*>(&_Mtx_storage + 1) - 1);
```

We only really need one `reinterpret_cast` at least.

Reply:
Great catch. As a stylistic variant, `*(CAST - 1)` could written as `CAST[-1]` - we used negative subscripting in similar situations:

https://github.com/microsoft/STL/blob/d4f1e09e34eef93c1fcbb750257e0ec8df2b8ff3/stl/inc/xmemory#L137

I am fine with either form.